src/catkin_lint/ros.py

```diff
@@ -54,12 +54,43 @@ def has_key(self, name):
         return self.view is not None and name in self.view.keys()
 
 
-def get_rosdep(quiet):
+def get_rosdep(quiet, sources_list_dir=None, init=False, sources_cache_dir=None, update=False):
     from rosdep2.lookup import RosdepLookup
     from rosdep2.rospkg_loader import DEFAULT_VIEW_KEY
-    from rosdep2.sources_list import SourcesListLoader
+    from rosdep2.sources_list import SourcesListLoader, update_sources_list, SOURCE_PATH_ENV,\
+        get_default_sources_list_file
+    from rosdep2.main import command_init
+    from rosdep2.rosdistrohelper import PreRep137Warning
+
+    if sources_list_dir is not None:
+        if not os.path.exists(sources_list_dir):
+            os.makedirs(sources_list_dir)
+        os.environ[SOURCE_PATH_ENV] = sources_list_dir
+
+    if init and not os.path.exists(get_default_sources_list_file()):
+        # command_init() writes to stdout, so we silence it
+        try:
+            # Python 2
+            from StringIO import StringIO
+        except ImportError:
+            # Python 3
+            from io import StringIO
+        stdout = sys.stdout
+        sys.stdout = StringIO()
```

Comment:
Use[ `contextlib.redirect_stdout`](https://docs.python.org/3/library/contextlib.html#contextlib.redirect_stdout)

Reply:
I'd like to, but that's not available in Python 2.7 :(

Reply:
Are we still supporting Python 2? 😲

Reply:
Pretty well, actually :) Melodic isn't dead yet =)

But not in catkin_lint_cmake...

Reply:
No, I decided to not support Python 2 in a project I created significant time past the EOL of Python 2.

Reply:
That's completely okay you decided to abandon Melodic. And that's why I wanted to pull a (simpler) CMake support directly into catkin_lint.